x/bor/keeper/veblop.go

```diff
@@ -408,15 +409,16 @@ func (k *Keeper) FilterByActiveProducerSet(ctx context.Context, candidates []uin
 	return activeCandidates
 }
 
-// SelectProducer selects a producer from the candidates list.
-// The selected candidate will be the next candidate to the current producer.
-// If the current producer is not in the candidate list, the first candidate in the list will be chosen.
-func (k *Keeper) SelectProducer(ctx context.Context, currentProducer uint64, candidates []uint64) (uint64, error) {
+// SelectProducer selects a producer from the candidates list, starting after the current producer,
+// wrapping around. It skips any candidate that is down for the given [startBlock, endBlock].
+// If all candidates are down (full cycle), it returns candidates[0].
+func (k *Keeper) SelectProducer(ctx sdk.Context, currentProducer uint64, candidates []uint64, startBlock, endBlock uint64) (uint64, error) {
```

Comment:
Same as before with context? Just double checking as I remember we had some issues with contexts switches in the past in function signatures of keepers' methods